langchain4j/src/main/java/dev/langchain4j/service/ServiceOutputParser.java

```diff
@@ -115,34 +115,36 @@ public static String outputFormatInstructions(Class<?> returnType) {
             return "\nYou must put every item on a separate line.";
         }
 
-        return "\nYou must answer strictly in the following JSON format: " + jsonStructure(returnType);
+        return "\nYou must answer strictly in the following JSON format: " + jsonStructure(returnType, new HashSet<>());
     }
 
-    private static String jsonStructure(Class<?> structured) {
+    private static String jsonStructure(Class<?> structured, Set<Class<?>> visited) {
         StringBuilder jsonSchema = new StringBuilder();
-        jsonSchema.append("{\n");
+        String simpleTypeName = simpleTypeName(structured);
+        visited.add(structured);
+        jsonSchema.append(simpleTypeName + ": {\n");
```

Comment:
I'm curious: What's the reasoning for the class name before json structure?

Reply:
Hi, you can see it in ServiceOutputParserTest.outputFormatInstructions_PersonWithParentArray the PersonWithParentArray description is not repeated if used multiple time in your data structures.

Reply:
Understood, but my worry is that this it produces something that is not valid json, for example:

```
You must answer strictly in the following JSON format: org.acme.ClaimSummary: {\n\"title\": (type: string),\n\"summary\": (type: string),\n\"date\": (type: string),\n\"location\": (type: string),\n\"policyNumber\": (type: string),\n}
```

Reply:
the format 
"propertyLabel": (type: classname), ...
seems to be well understand by LLMs, a maybe more reliable way could be to use json-schema but it is a lot more verbose, we should consider usage with token based API pricing. As far as I can say it seems to be a good compromise

Reply:
Do you have any links explaining that using the name works? Asking because I've never seen this used before and it strikes me as odd that it works (and I did see it does, although my colleague @cescoffier ran into issues with it)

Reply:
It didn't work with GPT 3.5 and Mistral for me. I had to switch back to String, sanitize and do the mapping myself.

Reply:
oh maybe you are just talking about the first org.acme.ClaimSummary?
_You must answer strictly in the following JSON format: **org.acme.ClaimSummary:** {\n\"title\": (...._
If this is what you are talking about, this was a suggestion from @langchain4j also to save token. It seemed to work well so I had no further objection

Reply:
> oh maybe you are just talking about the first org.acme.ClaimSummary?

Yes exactly. This wasn't part of the prompt in previous versions and now its addition is breaking thing unfortunatelt

Reply:
mmmm no I have no reference to provide, if it break things we should revert this part before this commit : https://github.com/langchain4j/langchain4j/pull/625/commits/bbf8f67722bfeb33fe699875eaf11f4811affc85
Just waiting @langchain4j if he has some reference to provide and I will make a new PR

Reply:
@langchain4j I really think the class name should be removed from the JSON as @cescoffier is having issues with it with various LLMs

Reply:
Well I am not a big fan of having a class name in front of schema and I think I raised this concern before, but somehow it worked in ITs, so we went on. Maybe it makes sense to add it only when there is a cycle (self-reference).

But in any case, do you guys enable [json mode](https://docs.langchain4j.dev/tutorials/ai-services#json-mode) when doing POJO extraction?

Reply:
@geoand it would make sense to enable the json mode if the result is not a String?

Reply:
> the fix will be in #844

Thanks!

Reply:
When return type is POJO, we could politely ask `ChatLanguageModel` to use json mode, if it supports it.
To avoid forcing user to set json mode on the model explicitly.

Reply:
That would require overloading `generate()` methods to accept same parameters that are configured in the constructor of `ChatLanguageModel`. Long asked feature...

Reply:
@geoand availability of "JSON mode" feature depends more on the inference runtime than on the model itself. For example, Ollama supports it for all models.

Reply:
Sorry, don't get it. How to you want to "enable" it? In my understanding if return type is POJO and LLM provider supports JSON mode, library should automatically enable it, there is no harm in it.

Reply:
What happens if you enable it but you don't need a json response, you only need simple string response for example?

Reply:
I mean JSON mode will be enabled only for a single request to the LLM and only if the return type is POJO

Reply:
Right, we are saying the same thing I think: that the library can / should enable json mode when it knows that json needs to be returned.